leshan-tl-javacoap-server/src/main/java/org/eclipse/leshan/transport/javacoap/server/request/LwM2mResponseBuilder.java

```diff
@@ -117,14 +118,38 @@ public LwM2mResponseBuilder(CoapResponse coapResponse, CoapRequest coapRequest,
 
     @Override
     public void visit(ReadRequest request) {
+        ContentFormat contentFormat = coapResponse.options().getContentFormat() != null
+                ? ContentFormat.fromCode(coapResponse.options().getContentFormat())
+                : request.getContentFormat();
+
+        TimestampedLwM2mNode timestampedNode = null;
+        try {
+            List<TimestampedLwM2mNode> timestampedNodes = decoder.decodeTimestampedData(
+                    coapResponse.getPayload().getBytes(), contentFormat, request.getPath(), model);
+            timestampedNode = timestampedNodes != null ? timestampedNodes.get(0) : null;
+        } catch (CodecException e) {
+            LOG.debug(String.format("Unable to decode response payload of request [%s]", request));
+        }
+
```

Comment:
Probably better to create dedicated method here too and call it only in : 

// handle success response

section only